src/contexts/alert_handler.jsx

```diff
@@ -0,0 +1,32 @@
+import React, { createContext, useContext, useState } from "react";
+
+const AlertHandlerContext = createContext();
+
+export const AlertHandlerProvider = ({ children }) => {
+  const [alertState, setAlertState] = useState({
+    visible: false,
+    errorText: "",
+  });
+
+  const setAlert = (errorText) => {
+    setAlertState({ visible: true, errorText: errorText });
+  };
+
+  const closeAlert = () => {
+    setAlertState({ visible: false });
+  };
```

Comment:
```suggestion
  const closeAlert = () => {
    setAlertState({ ...alertState, visible: false });
  };
```
現状、アラートが消える際に中の文字がなくなってしまっています。
こうすると、既存のalertStateのエラーメッセージが保持されるので、見た目がより自然になるかと思います。